internal/engine/buildcontrol/docker_compose_build_and_deployer.go

```diff
@@ -79,23 +111,23 @@ func (bd *DockerComposeBuildAndDeployer) BuildAndDeploy(ctx context.Context, st
 		})
 	}()
 
-	q, err := NewImageTargetQueue(ctx, iTargets, currentState, bd.ib.CanReuseRef)
+	q, err := NewImageTargetQueue(ctx, plan.tiltManagedImageTargets, currentState, bd.ib.CanReuseRef)
 	if err != nil {
 		return store.BuildResultSet{}, err
 	}
 
-	numStages := q.CountBuilds()
+	// base number of stages is the Tilt-managed image builds + the Docker Compose up step (which might be launching
+	// a Tilt-built image OR might build+launch a Docker Compose-managed image)
+	numStages := q.CountBuilds() + 1
 
 	reused := q.ReusedResults()
 	hasReusedStep := len(reused) > 0
 	if hasReusedStep {
 		numStages++
 	}
 
-	haveImage := len(iTargets) > 0
-
 	ps := build.NewPipelineState(ctx, numStages, bd.clock)
-	defer func() { ps.End(ctx, err) }()
+	defer ps.End(ctx, err)
```

Comment:
this change isn't right -- i think you're confused about defer evaluation order

before this change, err gets evaluated at the end of the function.

after this change, err gets evaluated immediately.

Reply:
here's a good explanation of it: https://medium.com/@manandharsabbir/go-lang-defer-statement-arguments-evaluated-at-defer-execution-b2c4a1687c6c

Reply:
Whoops 🤦 good catch - fixed in 4d76c6e2e